src/xenia/kernel/xam/xam_nui.cc

```diff
@@ -35,23 +57,297 @@ struct X_NUI_DEVICE_STATUS {
 };
 static_assert(sizeof(X_NUI_DEVICE_STATUS) == 24, "Size matters");
 
-void XamNuiGetDeviceStatus_entry(pointer_t<X_NUI_DEVICE_STATUS> status_ptr) {
+// Get
+dword_result_t XamNuiGetDeviceStatus_entry(
+    pointer_t<X_NUI_DEVICE_STATUS> status_ptr) {
+  /* Notes:
+     - it does return a value that is not always used
+     - returns values are X_ERROR_SUCCESS, 0xC0050006, and others
+     - 1) On func start *status_ptr = 0, status_ptr->unk1 = 0, status_ptr->unk2
+     = 0, and status_ptr->status = 0
+     - 2) calls XamXStudioRequest(6,&var <- = 0);
+     - if return is greater than -1 && var & 0x80000000 != 0 then set
+     status_ptr->unk1 = 0xC000009D, status_ptr->unk2 = 0xC000009D, and
+     status_ptr->status = status_ptr[3] = 0x20
+     - lots of branching functions after
+  */
+
   status_ptr.Zero();
-  status_ptr->status = 0;  // Not connected.
+  status_ptr->status = cvars::NUI_Ready ? true : false;
+  return cvars::NUI_Ready ? X_ERROR_SUCCESS : 0xC0050006;
 }
 DECLARE_XAM_EXPORT1(XamNuiGetDeviceStatus, kNone, kStub);
 
+dword_result_t XamUserNuiGetUserIndex_entry(unknown_t unk, lpdword_t index) {
+  return X_E_NO_SUCH_USER;
+}
+DECLARE_XAM_EXPORT1(XamUserNuiGetUserIndex, kNone, kStub);
+
+dword_result_t XamUserNuiGetUserIndexForSignin_entry(lpdword_t index) {
+  return X_E_FAIL;
+}
+DECLARE_XAM_EXPORT1(XamUserNuiGetUserIndexForSignin, kNone, kStub);
+
+dword_result_t XamUserNuiGetUserIndexForBind_entry(lpdword_t index) {
+  return X_E_FAIL;
+}
+DECLARE_XAM_EXPORT1(XamUserNuiGetUserIndexForBind, kNone, kStub);
+
+dword_result_t XamNuiGetDepthCalibration_entry(lpdword_t unk1) {
+  /* Notes:
+     - Possible returns X_STATUS_NO_SUCH_FILE, and 0x10000000
+  */
+  return X_STATUS_NO_SUCH_FILE;
+}
+DECLARE_XAM_EXPORT1(XamNuiGetDepthCalibration, kNone, kStub);
+
+// Skeleton
+qword_result_t XamNuiSkeletonGetBestSkeletonIndex_entry(int_t unk) {
+  return 0xffffffffffffffff;
+}
+DECLARE_XAM_EXPORT1(XamNuiSkeletonGetBestSkeletonIndex, kNone, kStub);
+
+/* XamNuiCamera Notes
+   - most require message calls to xam in 0x0002Bxxx area
+*/
+dword_result_t XamNuiCameraTiltGetStatus_entry(lpdword_t unk) {
```

Comment:
This pointer contains 2 32bit fields. That 0x50 is amount of bytes to copy and second one is always set to: 0x58745373 "(XtSs)?"